### ISSUE TITLE
[Bug] Nested Stacklayout inside a Frame overlaps tab bar

### ISSUE THREAD

**Issue by luiseduardohd** [xamarin/Xamarin.Forms#14743]
### Description
Anything inside an StackLayout inside a frame overlaps with the tab bar.

### Steps to Reproduce

1. Create a tabbedpage
2. Add pages
3. Inside a page create a Frame and inside a frame an Stacklayout and a label with a big text.

### Expected Behavior
The text from the label should not overlap the tab bar

### Actual Behavior
The text from the label overlaps the tab bar on ios

### Basic Information

- Version with issue:
- Xamarin.Forms (5.0.0.2125)
- Last known good version: ?
- Platform Target Frameworks: 
  - iOS:  .Net Standard 2.0
- NuGet Packages:
- Xamarin.Forms (5.0.0.2125)
- Affected Devices:
- iPhone Simulator - iPhone 12 Pro iOS 15

### Screenshots
![simulator_screenshot_CC436569-129D-4B76-9554-64A2994382C4](https://user-images.githubusercontent.com/1307611/137649427-372d1eac-68e2-4229-a9a9-20b132363d5c.png)

### Reproduction Link
https://github.com/luiseduardohd/Xamarin_bug_Stacklayout_inside_a_frame

**Comment by jfversluis** [comment]
Yeah tbh it seems weird to me as well. So we'll have another look here, thanks! :)
